src/test/java/com/teragrep/cfe_39/CombinedFullTest.java

```diff
@@ -0,0 +1,309 @@
+            }
+        }
+
+    }
+    @Test
+    public void hdfsPruneTest() throws IOException {
+        // Check that the files were properly written to HDFS with a read test.
+        String hdfsuri = config.getHdfsuri();
+
+        String path = config.getHdfsPath()+"/"+"testConsumerTopic";
+        // ====== Init HDFS File System Object
+        Configuration conf = new Configuration();
+        // Set FileSystem URI
+        conf.set("fs.defaultFS", hdfsuri);
+        // Because of Maven
+        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        // Set HADOOP user
+        System.setProperty("HADOOP_USER_NAME", "hdfs");
+        System.setProperty("hadoop.home.dir", "/");
+        //Get the filesystem - HDFS
+        FileSystem fs = FileSystem.get(URI.create(hdfsuri), conf);
+
+        //==== Create directory if not exists
+        Path workingDir=fs.getWorkingDirectory();
+        Path newDirectoryPath= new Path(path);
+        if(!fs.exists(newDirectoryPath)) {
+            // Create new Directory
+            fs.mkdirs(newDirectoryPath);
+            LOGGER.info("Path {} created.", path);
+        }
+
+        // Use either HDFS-file modification timestamps or avro-mapred for pruning.
+
+        // The records are in this AVRO format:
+        // {"timestamp": 1650872092240000, "message": "25.04.2022 07:34:52.240 [WARN] com.teragrep.jla_02.Log4j2 [instanceId=01, thread=Thread-0, userId=, sessionId=, requestId=, SUBJECT=, VERB=, OBJECT=, OUTCOME=, message=Log4j2 warn audit says hi!]", "directory": "jla02logger", "stream": "test:jla02logger:0", "host": "jla-02.default", "input": "imrelp:cfe-06-0.cfe-06.default:", "partition": "8", "offset": 8, "origin": "jla-02.default"}
+        // Query handler must be implemented in a way that the AVRO files are first opened, then processed to syslog format and then sent to the query requester. The records are processed/filtered based on the given query conditions using MapReduce to make the code capable of processing the vast amounts of records that are expected.
+        // MapReduce functionalities of the Hadoop cluster: https://hadoop.apache.org/docs/stable/hadoop-mapreduce-client/hadoop-mapreduce-client-core/MapReduceTutorial.html
+        // Avro side of documentations for MapReduce: https://avro.apache.org/docs/1.11.1/mapreduce-guide/
+
+        // Another method for pruning aside using avro-mapred is to use modification timestamp of the avro-file stored in HDFS:
+        // fs.setTimes(new Path(path+"/"+0.8), Long.parseUnsignedLong("1675930598000"), -1);
+        // where mtime is modification time and atime is access time. -1 as input parameter leaves the original atime/mtime value as is.
+        FileStatus[] fileStatuses = fs.listStatus(new Path(newDirectoryPath + "/"));
+        long count = Arrays.stream(fileStatuses).count();
+        if (count != 0) {
```

Comment:
No ifs, use assertions to make the test fail if they are not at expected state

Reply:
Solved from commit 65342c2e865912f981dfe82f470cab6884f72cdb onward